src/test/java/bgu/spl/mics/SquadTest.java

```diff
@@ -1,19 +1,62 @@
 package bgu.spl.mics;
 
+import bgu.spl.mics.application.passiveObjects.Agent;
+import bgu.spl.mics.application.passiveObjects.Inventory;
+import bgu.spl.mics.application.passiveObjects.Squad;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
-import static org.junit.jupiter.api.Assertions.fail;
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.*;
 
 public class SquadTest {
-    @BeforeEach
-    public void setUp(){
 
+    public static final Agent agents[] = {new Agent("001", "a"), new Agent("002", "b"),
+            new Agent("003", "c"), new Agent("004", "d")};
+    public static final ArrayList<String> agentsNames = new ArrayList<>(Arrays.asList("a", "b", "c", "d"));
+    public static final ArrayList<String> agentsSerials = new ArrayList<>(Arrays.asList("001", "002", "003", "004"));
+    static Squad squad;
+
+    @BeforeAll
+    public static void setUp() {
+        squad = Squad.getInstance();
+        squad.load(agents);
+    }
+
+    @Test
+    public void testSingleTon(){
+        assertNotNull(squad, "singleTon not created");
+        Squad otherInstance = Squad.getInstance();
+        assertSame(squad, otherInstance, "different instances of singleton Squad exist!");
+    }
+
+    @Test
+    public void testLoad(){
+        ArrayList<String> testSerials = new ArrayList<>();
+        ArrayList<String> testNames = new ArrayList<>();
+        for (Agent agent: agents) {
+            testSerials.add(agent.getSerialNumber());
+            testNames.add(agent.getName());
+        }
```

Comment:
This isn't getting the agents you loaded to the Squad. It takes them from the list of agents you created in the beginning. This is maybe a test the agents weren't changed. But not a test for `load()`

Reply:
I'm checking that all serials in the squad has the appropriate name and vice versa,  and so I check existence and validity.

Reply:
no other way to do this because agents in squad is private..

Reply:
no other way to do this because agents in squad is private..

Reply:
But Agents is just the arraylist you created. It makes no sense.
On the other hand by using getAgents(serials) you would get the agents stored in the Squad object. Which is what you need.

Reply:
Ohhh ok
I was sure this what I do. Now I understand